async-nats/src/options.rs

```diff
@@ -819,11 +819,20 @@ impl ConnectOptions {
         self
     }
 
+    /// By default, client construction will fail if
+    /// the connection to the server cannot be established.
+    ///
+    /// Setting `retry_on_initial_connect` causes client construction
+    /// to succeed regardless: in case of initial connection failure,
+    /// the client will periodically attempt connection
+    /// (see [`ConnectOptions::event_callback`]).
     pub fn retry_on_initial_connect(mut self) -> ConnectOptions {
         self.retry_on_initial_connect = true;
         self
     }
 
+    /// By default, a server may advertise other servers in the cluster known to it.
+    /// By setting this option, a connection to these servers will not be attempted.
```

Comment:
Probably easier to state that client will ignore those advertised servers.
Also adding a short note why it could be useful would be nice - meaning that client might not be able to reach other servers advertised URLs for some reasons.